src/components/common/PrivateRoute/PrivateRoute.js

```diff
@@ -0,0 +1,12 @@
+import { useAuth } from 'providers/AuthProvider/AuthProvider';
+import { Navigate } from 'react-router-dom';
+
+const PrivateRoute = ({ children }) => {
+  const { user } = useAuth();
+  if (user) {
+    return children;
+  }
+  return <Navigate to="/signin" replace />;
```

Comment:
Why signin and not login 
Bcoz most of the website i come across they ask for login first

Reply:
Signin/Login both are synonyms for each other